synapsemonitor/monitor.py

```diff
@@ -113,6 +113,46 @@ def _find_modified_entities_file(syn: Synapse, syn_id: str, days: int = 1) -> li
     return []
 
 
+def _traverse(
+    syn: Synapse, synid_root: str, include_types: list = ["folder", "file", "project"]
+) -> list:
+    """Traverse Synapse entity hierarchy to gather all descendant
+    entities of a root entity.
+    Args:
+        syn: Synapse connection
+        synid_root: Synapse ID of root entity.
+        include_types: Must be a list of entity types (ie. [“folder”,”file”])
+            which can be found here:
+            http://docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityType.html
+    Returns:
+        List of descendant Synapse IDs and root Synapse ID
+    """
+
+    synid_desc = []
+    exclude_folder = True
+
+    # full traverse depends on examining folder entities, even if not requested
+    include_types_mod = include_types.copy()
+    for include_type in include_types:
+        if include_type == "folder":
+            exclude_folder = False
+    if exclude_folder:
+        include_types_mod.append("folder")
```

Comment:
```suggestion
    include_types_mod = set(include_types)
    if "folder" not in include_types_mod:
        include_types_mod.add("folder")
```

Reply:
I think this is cleaner, then you can remove exclude folder.  It seems like what you are saying here is that "folder" is always required.

Reply:
Removed the exclude_folder flag to clean it kept `include_type_mod` as a list as this is a argument feeding back into the recursion and therefore must be the same type as `include_type`